src/settings/templates/EditFieldPopup.tpl

```diff
@@ -245,18 +245,21 @@ function md_popupSave()
 	o_data.relatedModule				= o_data.UITypeNum == 10 ? $("*[name='related_modules\\[\\]']").map(function(){ldelim}return $(this).val();{rdelim}).get() : undefined;
 	o_data.addRelatedList				= o_data.UITypeNum == 10 ? $("*[name='add_related_list\\[\\]']").map(function(){ldelim}return $(this).is(":checked");{rdelim}).get() : undefined;
 	o_data.pickListValues				= o_data.UITypeNum == 15 || o_data.UITypeNum == 16 || o_data.UITypeNum == 33 ? $("input[name='picklist_values']").val() : undefined;
-		
+
 	//Not allow "name" as fieldname
 	if(o_data.fieldName == 'name')
 	{
 		alert(window.parent.md_vtranslate("LBL_VTIGER_DOESNT_LIKE_NAME"));
 		return false;
 	}
-	
+
 	//Not allow several fields as identifier
 	if(o_data.isEntityIdentifier)
-	{	
-		if(window.parent.md_entityIdentifier != undefined && window.parent.md_entityIdentifier != o_data.oldFieldName)
+	{
+		if(window.parent.md_entityIdentifier != undefined &&
+			window.parent.md_entityIdentifier.length > 0 &&
```

Comment:
This is the actual "meat" of the patch. The rest is just my basic Sublime Text settings correcting some whitespace trailing around.